snooty/parser.py

```diff
@@ -365,7 +364,20 @@ def handle_directive(
             doc["argument"] = argument
 
         argument_text = None
+        expected_num_columns = 0
         try:
+            if name == "list-table":
+                # Calculate the expected number of columns for this list-table structure.
```

Comment:
Try getting rid of the custom `ListTableDirective` directive handler below; then `widths` should be available via the `options` variable so you don't need to do all this string manipulation.